Paint/Paint/MainForm.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.IO;
+using System.Windows.Forms;
+
+namespace Paint
+{
+    public partial class MainForm : Form
+    {
+        private Pen drawingPen;
+        private Color currentPenColor;
+        private float penWidth;
+        private bool isPenDown;
+        private Point lastPoint = Point.Empty;
+        private Image lastImage;
+        public MainForm()
+        {
+            InitializeComponent();
+            ColorPickerBtn.Click += SetNewColor;
+            NewImageBtn.Click += CreateNewImage;
+            ClearBtimapBtn.Click += ClearImage;
+            OpenImageBtn.Click += OpenOpeningImageDialog;
+            SaveImageBtn.Click += OpenSavingImageDialog;
+            
+            CanvasPB.MouseMove += CoordinatesSetter;
+            CanvasPB.MouseMove += DrawByPressedPen;
+            CanvasPB.MouseDown += OnMouseDown;
+            CanvasPB.MouseUp += OnMouseUp;
+            CanvasPB.MouseLeave += CoordinatesClearer;
+            
+            currentPenColor = Color.Black;
+            penWidth = 10;
+            PenSizeUD.Value = (decimal) penWidth;
+            ColorPickerBtn.BackColor = currentPenColor;
+            drawingPen = new Pen(currentPenColor, penWidth);
+            drawingPen.StartCap = drawingPen.EndCap = LineCap.Round;
+            lastImage = new Bitmap(607, 542);
+            
+            PenSizeUD.ValueChanged += UpdatePen;
+            UpdateImage();
+        }
+
+        private void CreateNewImage(object sender, EventArgs eventArg)
+        {
+            var newBitmapScale = ShowCustomDialog();
+            lastImage = new Bitmap(newBitmapScale.X, newBitmapScale.Y);
+            UpdateImage();
```

Comment:
В это участке кода возникает ошибка выполнения, если хотя бы один из размеров нулевой. Это, например, получается, если нажать кнопку "New", а затем закрыть окно диалога крестиком. Нужна проверка
```suggestion
            if(newBitmapScale.X != 0 && newBitmapScale.Y != 0)
            {
                lastImage = new Bitmap(newBitmapScale.X, newBitmapScale.Y);
                UpdateImage();
            }
```